gobblin-docs/sources/QueryBasedSource.md

```diff
@@ -0,0 +1,92 @@
+[TOC]
+
+# Introduction
+[`QueryBasedSource`](https://github.com/linkedin/gobblin/blob/master/gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java)
+represents a category of sources whose data is pulled by sending queries. A dataset of A source is identified as a
+[`SourceEntity`](https://github.com/linkedin/gobblin/blob/master/gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedSource.java#L96).
+Query can be done by sending HTTP requests or SQL commands. A source often, not always, has a corresponding
+[`QueryBasedExtractor`](https://github.com/linkedin/gobblin/blob/master/gobblin-core/src/main/java/gobblin/source/extractor/extract/QueryBasedExtractor.java),
+which defines the way and implements common routines to extract data from the source.
+
+# Constructs
+## `QueryBasedSource`
+<p align="center">
+  <figure>    
+    <img src="/img/Gobblin-Query-Based-Sources.png" alt="Query based sources" width="800">
+    <figcaption><br>Figure 1: Query based sources<br></figcaption>
+  </figure>
+</p>
+
+Like other categories of sources, a `QueryBasedSource` focuses on creating work units as well. The way it does follows the general pattern:
+
+- calculate low watermark of current run based on previous runs
+- compute a high watermark
+- partition datasets of current run into work units
+- pick up peviously failed work units.
+
+At last, it will group several work units as
+[`MultiWorkUnit`](https://github.com/linkedin/gobblin/blob/master/gobblin-api/src/main/java/gobblin/source/workunit/MultiWorkUnit.java)
+according to the `mr.job.max.mappers` configuration.
```

Comment:
This should be specific to mr mode. Can you check if you do any bin-packing for local mode as well?